lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1340,6 +1351,13 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
                 setVariedIfDifferentiable(dirRes, i);
             }
           }
+          auto arg = ai->getCallee();
```

Comment:
It'd be nice not to repeat this logic. Can you merge this with the loop on line 1346?